src/util/compileProject.js

```diff
@@ -165,7 +165,37 @@ async function addJavascript(
     return;
   }
 
-  let source = `\n${sourceDelimiter}\n${javascript}`;
+  const [babel, presetEnv] = await retryingFailedImports(
+    () => Promise.all([
+      import(
+        /* webpackChunkName: "mainAsync" */
+        '@babel/core',
+      ),
+      import(
+        /* webpackChunkName: "mainAsync" */
+        '@babel/preset-env',
+      ),
+    ]),
+  );
+
+  let transformedCode = javascript;
+
+  try {
+    transformedCode = await babel.transformAsync(javascript, {
+      presets: [
+        [
+          presetEnv,
+          {
+            targets: 'chrome > 23',
+          },
+        ],
+      ],
+    });
+  } catch (e) {
+    return;
```

Comment:
We are catching syntax errors here before outputting it- this is probably not the ideal behavior, but as I write tests I'll flesh out the failure case

Reply:
hmm, I think the validator should detect all syntax errors? and if there is a static validation error then we don’t run `compileProject` at all? might be some nuance i am missing tho